src/alert/lsst.rs

```diff
@@ -445,28 +483,62 @@ pub struct DiaForcedSource {
     pub mjd: f64,
     /// Filter band this source was observed with.
     pub band: Option<String>,
+}
 
-    // these fields are computed later, from the psf_flux
+#[derive(Debug, PartialEq, Clone, serde::Deserialize, serde::Serialize)]
+pub struct ForcedPhot {
+    #[serde(flatten)]
+    pub dia_forced_source: DiaForcedSource,
     pub magpsf: Option<f32>,
     pub sigmapsf: Option<f32>,
     pub diffmaglim: Option<f32>,
     pub isdiffpos: Option<bool>,
     pub snr: Option<f32>,
 }
 
+impl TryFrom<DiaForcedSource> for ForcedPhot {
+    type Error = AlertError;
+    fn try_from(dia_forced_source: DiaForcedSource) -> Result<Self, Self::Error> {
+        let flux = dia_forced_source
+            .psf_flux
+            .ok_or(AlertError::MissingFluxPSF)?
+            * 1e-9;
+        let flux_err = dia_forced_source
+            .psf_flux_err
+            .ok_or(AlertError::MissingFluxPSF)?
+            * 1e-9;
+
+        let (magpsf, sigmapsf) = flux2mag(flux.abs(), flux_err);
+        let diffmaglim = fluxerr2diffmaglim(flux_err);
+
+        Ok(ForcedPhot {
+            dia_forced_source,
+            magpsf: Some(magpsf),
+            sigmapsf: Some(sigmapsf),
+            diffmaglim: Some(diffmaglim),
+            isdiffpos: Some(flux > 0.0),
+            snr: Some(flux.abs() / flux_err),
```

Comment:
Do you still want these fields to be `Option`s?

Reply:
No you're right, these don't need to be options, I'll change that

Reply:
done